src/combinatorics/simple_graph/clique.lean

```diff
@@ -68,4 +80,91 @@ variables {G H}
 lemma is_n_clique.mono (h : G ≤ H) : G.is_n_clique n s → H.is_n_clique n s :=
 by { simp_rw is_n_clique_iff, exact and.imp_left (is_clique.mono h) }
 
+@[simp] lemma is_n_clique_bot_iff : (⊥ : simple_graph α).is_n_clique n s ↔ n ≤ 1 ∧ s.card = n :=
+begin
+  rw [is_n_clique_iff, is_clique_bot_iff],
+  refine and_congr_left _,
+  rintro rfl,
+  exact card_le_one.symm,
+end
+
+end n_clique
+
+/-! ### Graphs without cliques -/
+
+section clique_free
+variables {n : ℕ}
+
+/-- `G.clique_free n` means that `G` has no `n`-cliques. -/
+def clique_free (n : ℕ) : Prop := ∀ t, ¬ G.is_n_clique n t
+
+variables {G H}
+
+lemma clique_free_bot (h : 2 ≤ n) : (⊥ : simple_graph α).clique_free n :=
+begin
+  rintro t ht,
+  rw is_n_clique_bot_iff at ht,
+  linarith,
+end
+
+lemma clique_free.mono (h : G ≤ H) : H.clique_free n → G.clique_free n :=
+forall_imp $ λ s, mt $ is_n_clique.mono h
+
+end clique_free
+
+/-! ### Finset of cliques -/
+
+section clique_finset
+variables (G) [fintype α] [decidable_eq α] [decidable_rel G.adj] {n : ℕ} {a b c : α} {s : finset α}
+
+/-- The `n`-cliques in a graph as a finset. -/
+def clique_finset (n : ℕ) : finset (finset α) := univ.filter $ G.is_n_clique n
```

Comment:
Why is it `clique_finset n : finset (finset α)` instead of `clique_set n : set (finset α)`? It doesn't look like anything is using the fact that `α` is a `fintype` or that there are finitely many cliques. Just looking for some justification.

Reply:
We will talk about `(G.clique_finset 3).card` later.